REST API: Show alert for inaccessible login or admin URL

itsmeichigo opened the pull request:
<!-- Remember about a good descriptive title. -->

Part of #8453 
<!-- Id number of the GitHub issue this PR addresses. -->

## Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
This PR updates the `DefaultApplicationPasswordUseCase` to check for the unauthorized error when generating application passwords (error code 401). This error can occur if the login or admin URL is not accessible, so we can listen to this error and show an alert about it on the UI.

## Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->
Prerequisite: Update your site login URL by installing the [Change wp-admin login URL](https://wordpress.org/plugins/change-wp-admin-login/) plugin and following its instructions.
- Enable the feature flag `applicationPasswordAuthenticationForSiteCredentialLogin` and build the app.
- Log out of the app if needed.
- On the prologue screen, select "Enter your site address".
- Enter your test store address and select "Sign in with site credentials".
- Enter the correct credentials for a user account with an eligible role and select "Continue". 
- Notice that an alert will then be displayed mentioning a custom login or admin URL causing application password generation to fail.
- Select the Contact Support button, you should be navigated back to the site address login screen and the support flow should start.

## Screenshots
<!-- Include before and after images or gifs when appropriate. -->
<img src="https://user-images.githubusercontent.com/5533851/210493732-282c82cf-47e6-4f6a-bd8d-fb76c307ff4c.png" width=320 />


---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

itsmeichigo committed:
Add a check for unauthorized request when generating application password

itsmeichigo committed:
Show specific alert for inaccessible login url

itsmeichigo committed:
Fix typo for error message

itsmeichigo committed:
Update unit tests for DefaultApplicationPasswordUseCase

selanthiraiyan commented on Networking/Networking/ApplicationPassword/ApplicationPasswordUseCase.swift:
```diff
@@ -160,6 +161,8 @@ private extension DefaultApplicationPasswordUseCase {
                         continuation.resume(throwing: ApplicationPasswordUseCaseError.applicationPasswordsDisabled)
                     case .responseValidationFailed(reason: .unacceptableStatusCode(code: ErrorCode.duplicateNameErrorCode)):
                         continuation.resume(throwing: ApplicationPasswordUseCaseError.duplicateName)
+                    case .responseValidationFailed(reason: .unacceptableStatusCode(code: ErrorCode.unauthorized)):
```
Thanks for covering this scenario!

selanthiraiyan reviewed (commented):
> Select the Contact Support button, you should be navigated back to the site address login screen and the support flow should start.

I am afraid the support flow doesn't start. 

![SupportFlowNotStarted](https://user-images.githubusercontent.com/524475/210541907-baff58c0-3e72-449c-a8f4-a82beba13dc3.gif)

itsmeichigo committed:
Remove deauthenticate call when selecting support

itsmeichigo commented:
> I am afraid the support flow doesn't start.

I added the `deauthenticate` call when handling the contact support action, but I forgot that it could cause the app to redirect to the home screen. I removed it in eff1bfa6baa956be72ef993b0391b8118d4f00b0, please take another look, thanks!

selanthiraiyan reviewed (approved):
This works as expected. 🎉 🚢
